scipy/integrate/_cubature.py

```diff
@@ -19,23 +20,27 @@
 
 @dataclass
 class CubatureRegion:
-    estimate: np.ndarray
-    error: np.ndarray
-    a: np.ndarray
-    b: np.ndarray
+    estimate: object
+    error: object
+    a: object
+    b: object
+    _xp: object
```

Comment:
Maybe use this like in `_lib._array_api`:

```python
from types import ModuleType

if TYPE_CHECKING:
    Array = Any  # To be changed to a Protocol later (see array-api#589)
    ArrayLike = Array | npt.ArrayLike

estimate: Array
...
b: Array
_xp: ModuleType
```

Reply:
Thanks for taking a look! I've had to add an `else: Array = object` since otherwise there's errors about `Array` not being defined when defining the dataclass if `TYPE_CHECKING = False`.